### PR TITLE
Fix Get-AzureRmSubscription issue in CloudShell

### PR DESCRIPTION
**Pull request by maddieclayton**
## Description
https://github.com/Azure/azure-powershell/issues/7360

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
sign job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/229/

File: src/ResourceManager/Profile/Commands.Profile/Subscription/GetAzureRMSubscription.cs [hunk 1]
```diff
@@ -104,8 +104,24 @@ public override void ExecuteCmdlet()
             {
                 try
                 {
-                    var subscriptions = _client.ListSubscriptions(tenant);
-                    WriteObject(subscriptions.Select((s) => new PSAzureSubscription(s)), enumerateCollection: true);
+                    if (DefaultContext.Account.Type.Equals("ManagedService"))
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Subscription/GetAzureRMSubscription.cs]: Please file an issue to add some automation for this - we have the infrastructure to mock managed service tests, and we should have some regression protection for thsi scenario.

Please make sure you have tested this in CloudShell.  LMK if you need some help in doign this.

**Comment by markcowl** [comment]
@maddieclayton Please file the issue for automated mocked tests for this

**Comment by maddieclayton** [comment]
Test coverage issue here: https://github.com/Azure/azure-powershell/issues/7536

I did test is on CloudShell and everything worked as expected.

**Pull request merged**